# Desafio:

Fazer uma analise detalhada dos smartphones mais vendidos do mercado, com isso ver quais são os detalhes mais importantes que levam o cliente a realizar a compra do aparelho.

# O que temos:

Temos 1 base de dados com informações dos smartphones e suas marcas, contendo também suas especificações

Download da Base de Dados: https://www.kaggle.com/datasets/juanmerinobermejo/smartphones-price-dataset

Referência: https://www.kaggle.com/sakshigoyal7/credit-card-customers

# Etapas:

Passo 1: Importar a Base de Dados

Passo 2: Visualizar e Tratar a Base de Dados

Passo 3: Analisar as informações dos dados.

Passo 4: Construir a Análise para identificar os principais motivos para a compra dos aparelhos.

- Identificar as principais características que levam determinados aparelhos a serem mais vendidos

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
# Passo 1: Importar a Base de Dados
import pandas as pd

tabela = pd.read_csv("/content/drive/MyDrive/smartphones.csv")

# Passo 2: Visualizar e Tratar a Base de Dados
tabela = tabela.drop("Free", axis=1)

display(tabela)

Smartphone     Brand  \
0                   Realme C55 8/256GB Sunshower Libre    Realme   
1             Samsung Galaxy M23 5G 4/128GB Azul Libre   Samsung   
2         Motorola Moto G13 4/128GB Azul Lavanda Libre  Motorola   
3             Xiaomi Redmi Note 11S 6/128GB Gris Libre    Xiaomi   
4              Nothing Phone (2) 12/512GB Blanco Libre   Nothing   
...                                                ...       ...   
1811             Xiaomi Redmi Note 8 4/64GB Azul Libre    Xiaomi   
1812   Xiaomi Redmi Note 8T 4/128GB Azul Estelar Libre    Xiaomi   
1813          Xiaomi Redmi Note 9 4/128GB Blanco Libre    Xiaomi   
1814   Xiaomi Redmi Note 9S 4/64GB Dual SIM Gris Libre    Xiaomi   
1815  Xiaomi Redmi Note 9S 6/128GB Dual SIM Gris Libre    Xiaomi   

               Model   RAM  Storage   Color  Final Price  
0                C55   8.0    256.0  Yellow       231.60  
1         Galaxy M23   4.0    128.0    Blue       279.00  
2           Moto G13   4.0    128.0    Blue       179.01  
3     Redmi Note 11S   6.0    128.0    Gray       279.99  
4          Phone (2)  12.0    512.0   White       799.00  
...              ...   ...      ...     ...          ...  
1811    Redmi Note 8   4.0     64.0    Blue       249.01  
1812   Redmi Note 8T   4.0    128.0    Blue       200.00  
1813    Redmi Note 9   4.0    128.0   White       269.00  
1814         Note 9S   4.0     64.0    Gray       211.00  
1815         Note 9S   6.0    128.0    Gray       289.00  

[1816 rows x 7 columns]

In [ ]:
# Passo 3: Analisar as informações dos dados.

tabela = tabela.dropna()
display(tabela.info())
display(tabela.describe().round(2))

<class 'pandas.core.frame.DataFrame'>
Index: 1333 entries, 0 to 1815
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Smartphone   1333 non-null   object 
 1   Brand        1333 non-null   object 
 2   Model        1333 non-null   object 
 3   RAM          1333 non-null   float64
 4   Storage      1333 non-null   float64
 5   Color        1333 non-null   object 
 6   Final Price  1333 non-null   float64
dtypes: float64(3), object(4)
memory usage: 83.3+ KB


None

RAM  Storage  Final Price
count  1333.00  1333.00      1333.00
mean      5.96   134.04       380.27
std       2.67    84.39       295.46
min       1.00    16.00        70.68
25%       4.00    64.00       189.00
50%       6.00   128.00       281.60
75%       8.00   128.00       475.74
max      12.00   512.00      2271.28

In [ ]:
qtde_marca = tabela["Brand"].value_counts()
display(qtde_marca)

Brand
Xiaomi        347
Samsung       307
Realme        117
OPPO           92
POCO           67
Motorola       52
Huawei         46
TCL            35
Cubot          34
Ulefone        29
Vivo           27
Honor          26
Blackview      26
Hammer         21
OnePlus        18
ZTE            15
Nokia          12
SPC             9
Google          9
Nothing         9
Alcatel         7
Crosscall       6
CAT             5
Gigaset         3
Qubo            3
Asus            3
Doro            2
Microsoft       1
Funker          1
LG              1
Maxcom          1
Swissvoice      1
Lenovo          1
Name: count, dtype: int64

In [ ]:
qtde_modelo = tabela["Model"].value_counts()
print(qtde_modelo)
print("-----------------------")
qtde_ram = tabela["RAM"].value_counts()
print(qtde_ram)
print("-----------------------")
qtde_storage = tabela["Storage"].value_counts()
print(qtde_storage)
print("-----------------------")
qtde_color = tabela["Color"].value_counts()
print(qtde_color)

Model
Redmi Note 12    37
Galaxy S20       33
Galaxy A13       26
11 Pro           26
Redmi Note 10    21
                 ..
G11               1
Zenfone 7         1
A53s              1
A57               1
Pocophone F2      1
Name: count, Length: 341, dtype: int64
-----------------------
RAM
8.0     381
4.0     380
6.0     259
3.0     112
12.0    109
2.0      88
1.0       4
Name: count, dtype: int64
-----------------------
Storage
128.0    614
256.0    274
64.0     260
32.0     150
16.0      19
512.0     16
Name: count, dtype: int64
-----------------------
Color
Black        416
Blue         294
Gray         171
Green        136
White        120
Silver        45
Purple        32
Orange        29
Gold          24
Red           19
Pink          19
Yellow        12
Brown          4
Turquoise      4
Crystal        4
Bronze         3
Rainbow        1
Name: count, dtype: int64


In [ ]:
# Passo 4: Construir a Análise para identificar os principais motivos para a compra dos aparelhos.

import plotly.express as px

for coluna in tabela:
  grafico = px.histogram(tabela, x=coluna, color="Brand")

  grafico.show()

# Analise completa dos dados

## Dados Descobertos

- Preferências de Marca:

  - A maioria dos clientes prefere as marcas Samsung e Xiaomi.
Essas marcas têm mais vendas e predominância no mercado devido à maior variedade de aparelhos lançados em comparação com os concorrentes.

- Faixa de Preço:

  - A principal faixa de valor para compras está entre 150 a 250     dólares.

- Modelos Mais Vendidos:

  - Os dois modelos mais vendidos são Redmi Note 12 e Samsung S20.

- Preferência de Memória RAM:

  - A preferência dos clientes é por memória RAM entre 4 a 8 GB, com 8 GB sendo a mais vendida.

- Armazenamento:

  - A maior escolha de armazenamento é de 128 GB, predominando fortemente no mercado.


## Plano de Ação Estratégica

- Diversidade de Produtos:

  - As empresas que não vendem tantos aparelhos devem diversificar a linha de smartphones para alcançar todos os tipos de público.

- Cores de Smartphones:

  - Optar por cores mais escuras, como preto, azul ou cinza, tende a aumentar o interesse dos clientes.

- Armazenamento:

  - Lançar smartphones com armazenamento de 128 GB é uma excelente estratégia, já que essa capacidade é a mais escolhida pelos clientes.

- Memória RAM:

  - Smartphones com memória RAM abaixo de 4 GB e acima de 8 GB não têm uma alta taxa de vendas. Portanto, manter a oferta principalmente entre 4 e 8 GB é ideal.
- Faixa de Preço:

  - Considerando que a faixa de preço preferida é entre 150 a 250 dólares, é aconselhável evitar ultrapassar os 400 dólares para manter uma boa taxa de vendas.